### PR TITLE
[YUNIKORN-1856] Pod creation fails in certain cases when generateUniqueAppId is enabled

### PR DESCRIPTION
**Pull request by mitdesai**
### What is this PR for?
Fix for podUid not available.

### What type of PR is it?
* [X] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
* Open an issue on Jira https://issues.apache.org/jira/browse/YUNIKORN-1856

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/admission/util_test.go [hunk 1]
```diff
@@ -263,18 +263,16 @@ func TestGenerateAppID(t *testing.T) {
 		conf.AMFilteringGenerateUniqueAppIds: fmt.Sprintf("%t", true),
 	})
 
-	podUid := "abcd1234-5678-efgh-90ij-klmnopqrstuv"
-
 	// short namespace name
 	ns := "short"
-	appID = generateAppID(ns, createTestingPodWithCustomFields(ns), uniqueEnabled.GetGenerateUniqueAppIds())
```
Comment (wilfred-s) [inline pkg/admission/util_test.go:270]: This leaves the function `createTestingPodWithCustomFields()` unused and we should remove it.

**Review by wilfred-s — CHANGES_REQUESTED** [review]

File: pkg/admission/util_test.go (hunk 1 above)
Comment (wilfred-s) [reply pkg/admission/util_test.go:270]: The lint check will probably fail due to this unused function

**Review by wilfred-s — APPROVED** [review]
LGTM, waiting for a build run to pass before committing.

**Comment by mitdesai** [comment]
Thanks for the review @wilfred-s

**Pull request closed without merging**